### PR TITLE
DevEx - DynamoDB Refactoring

### PR DESCRIPTION
**Pull request by codyseibert**
- adding a GSI so we can query by workitem id for updating them when a case status or docket number changes
- changing the billing mode to on-demand so we don't need to worry about what our RCU / WCU are set to
- storing the workitem id in the `gsi1pk` key for work items
- no longer using mapping records to fetch the work queues
- removing some unused functions

### CODE REVIEW

**Review by rachaelparris — APPROVED** [review]

File: shared/src/business/test/createMockDocumentClient.js [hunk 1]
```diff
@@ -55,12 +55,27 @@ const createMockDocumentClient = () => {
...
+          if (data[key].gsi1pk === gsi1pk) {
+            arr.push(data[key]);
+          }
+        } else {
+          const value = ExpressionAttributeValues[':pk'];
+          const prefix = ExpressionAttributeValues[':prefix'];
+          const [pk, sk] = key.split(' ');
+
+          if (prefix) {
+            // console.log('we are here', prefix, sk);
```
Comment (kkoskelin) [inline shared/src/business/test/createMockDocumentClient.js]: can be removed?

**Review by kkoskelin — APPROVED** [review]

**Review by wbyoko — APPROVED** [review]

**Pull request merged**